### PR TITLE
Enable warnings during D compilation and fix all warnings

### PR DESCRIPTION
**Pull request by JohanEngelen**
I have tried to fix all warnings in the way that seemed most "correct" to me.

See forum discussion here: http://forum.dlang.org/thread/xkftumtlbvmsrjrkedbh@forum.dlang.org

tested wih dmd 2.069.0 and 2.070.0

### CODE REVIEW

**Comment by JohanEngelen** [comment]
While working on this, I stumbled on what appears to be a bug: https://issues.dlang.org/show_bug.cgi?id=15731

File: src/posix.mak [hunk 1]
```diff
@@ -146,6 +146,8 @@ CXXFLAGS += \
 endif
 # Default D compiler flags for all source files
 DFLAGS=
+# Enable D warnings
+DFLAGS += -w
```
Comment (yebblies) [inline src/posix.mak]: `wi`

Comment (JohanEngelen) [inline src/posix.mak]: thanks, fixed.

**Comment by JohanEngelen** [comment]
@yebblies Fixed -wi, squashed and rebased on master, looked green on all testers.

**Pull request merged**